core/core_test.go

```diff
@@ -224,3 +226,30 @@ func TestDefaultAPI_ServiceCredentials(t *testing.T) {
 		})
 	}
 }
+
+func TestMaxBytes(t *testing.T) {
```

Comment:
I would suggest to add tests for edge conditions. This would detect the off-by-one mentioned in other comment. Generally a lot of bugs happen at the edge conditions that's why it's super useful to have edge conditions tested if possible

Reply:
done